anesthetic/samples.py

```diff
@@ -154,34 +160,39 @@ def plot(self, ax, paramname_x, paramname_y=None, *args, **kwargs):
                 xmin, xmax = self._limits(paramname_x)
                 ymin, ymax = self._limits(paramname_y)
                 if plot_type == 'kde':
+                    x = self[paramname_x]
+                    y = self[paramname_y]
+                    return kde_contour_plot_2d(ax, x, y, weights=self.w,
+                                               xmin=xmin, xmax=xmax,
+                                               ymin=ymin, ymax=ymax,
+                                               *args, **kwargs)
+                elif plot_type == 'fastkde':
                     x = self[paramname_x].compress()
                     y = self[paramname_y].compress()
```

Comment:
This concerns fastkde, so is actually less relevant considering that kde is taking over, but still. For large amounts of data, the contour plots with fastkde  have too much fine grained structure:

![image](https://user-images.githubusercontent.com/18258042/62838938-d2317780-bc7a-11e9-8d5a-8b7b2bd4d2ee.png)

Is `.compress()` geared towards the 1D case? Should we set a maximum `nsamples`? Any other idea of improving the compression in this case?

Reply:
I'm not sure why fastkde would be showing this behaviour, since it's supposed to be optimised for the case of lots of samples.

However, the `compress()` function does end up having repeated rows, for which fastkde may not be optimised for. What happens to the above plot if you change `compress()` for `compress(-1)` in `MCMCSamples.plot_2d` ?

Reply:
Not as chaotic but super wobbly:

![image](https://user-images.githubusercontent.com/18258042/62856251-4c4e1480-bcec-11e9-8630-98fad87e459a.png)

Reply:
It gets a bit better by manually reducing the number of samples. Here for `nsamples=10000`:

![image](https://user-images.githubusercontent.com/18258042/62858641-b9fd3f00-bcf2-11e9-80bc-65d548ee46af.png)

Maybe worth turning `nsamples` into a kwarg, then again maybe not that needed as we have `gaussian_kde` soon.

Reply:
OK. The latest push adds ncompress as a kwarg, which should give the control you desire for any of the plotting functions.